### PR TITLE
fix #2555: add backreference in CQs

### LINKED ISSUE #2555: [feature request] backreference for measurement/series when making CQs
I try to create downsampling series:

```
> create continuous query "1_mins_rollup" on collectd begin select mean(value) into "1_week".:series_name from /^[a-z]+.*/ group by time(1m) end
ERR: error parsing query: found :, expected identifier at line 1, char 92
```

It seems regex is not supported in influxdb 0.9 now. Then I have to create a continuous query for `every`  measurement or series. It is too tedious.

### PR DESCRIPTION
**Pull request by dgnorton**
Add new query syntax to allow the following in CQs:
```
INTO "1hPolicy".:MEASUREMENT FROM /.*/
```
In this example, the regular expression in the `FROM` clause would expand to all measurements and the points selected from each of those measurements would be written into the `"1hPolicy"` retention policy with the same measurement names.

### CODE REVIEW

**Comment by otoolep** [comment]
Looks like Circle blew up due to memory pressure. I've seen this before, and it can happen if the code allocates huge buffers by mistake, perhaps in a loop.

File: influxql/parser.go [hunk 1]
```diff
@@ -699,6 +702,8 @@ func (p *Parser) parsePrivilege() (Privilege, error) {
 // parseSelectStatement parses a select string and returns a Statement AST object.
 // This function assumes the SELECT token has already been consumed.
 func (p *Parser) parseSelectStatement(tr targetRequirement) (*SelectStatement, error) {
+	fmt.Println("parseSelectStatement: start")
```
Comment (pauldix) [inline influxql/parser.go]: forgot to take this out :)

File: influxql/parser.go [hunk 2]
```diff
@@ -699,6 +702,8 @@ func (p *Parser) parsePrivilege() (Privilege, error) {
...
+	defer fmt.Println("parseSelectStatement: end")
```
Comment (pauldix) [inline influxql/parser.go]: and this

File: services/continuous_querier/service_test.go [hunk 3]
```diff
@@ -61,6 +62,45 @@ func TestExecuteContinuousQuery_HappyPath(t *testing.T) {
...
+	}
+
+	err := s.ExecuteContinuousQuery(&dbi, &cqi)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
 // Test the service happy path.
 func TestService_HappyPath(t *testing.T) {
```
Comment (pauldix) [inline services/continuous_querier/service_test.go]: unrelated to this PR, but why is this test named this? `TestService` should actually be something like `TestContinuousQueryService` and `HappyPath`?

File: influxql/parser.go (hunk 1 above)
Comment (dgnorton) [inline influxql/parser.go]: bah!!  good catch.

**Comment by pauldix** [comment]
Can you add a test to ensure that `GROUP BY *` works with this as well to bring over the tag data? Would actually like to see more integration level tests if possible.

**Comment by otoolep** [comment]
@dgnorton -- did you work out what caused Circle to fail due to memory pressure?

File: services/continuous_querier/service.go [hunk 4]
```diff
@@ -346,7 +351,11 @@ func (s *Service) runContinuousQueryAndWriteResult(cq *ContinuousQuery) error {
 	}
 
 	if s.loggingEnabled {
-		s.Logger.Printf("wrote %d point(s) to %s.%s.%s", len(points), cq.intoDB(), cq.intoRP(), cq.Info.Name)
+		db := cq.intoDB()
```
Comment (otoolep) [inline services/continuous_querier/service.go]: This is purely a logging thing?

**Comment by dgnorton** [comment]
@otoolep I added a `Parent Node` field to `type Measurement` which caused `json.Marshal` call in the tests to go into an infinite loop.

**Comment by dgnorton** [comment]
@pauldix added integration tests but issue #3968 prevents the `GROUP BY *` test from passing, so I've disabled that test until the issue is fixed.

File: cmd/influxd/run/server_test.go [hunk 5]
```diff
@@ -3643,37 +3643,111 @@ func TestServer_Query_CreateContinuousQuery(t *testing.T) {
...
+	// Setup tests to check the CQ results.
+	test2 := NewTest("db0", "rp1")
+	test2.addQueries([]*Query{
+		&Query{
+			name:    "check results of cq1",
+			command: `SELECT * FROM "rp1"./[cg]pu/`,
+			exp:     `{"results":[{"series":[{"name":"cpu","columns":["time","count","host","region","value"],"values":[["` + interval2.UTC().Format(time.RFC3339Nano) + `",3,null,null,null]]},{"name":"gpu","columns":["time","count","host","region","value"],"values":[["` + interval1.UTC().Format(time.RFC3339Nano) + `",2,null,null,null],["` + interval0.UTC().Format(time.RFC3339Nano) + `",1,null,null,null]]}]}]}`,
+			params:  url.Values{"db": []string{"db0"}},
+		},
+		// TODO: restore this test once this is fixed: https://github.com/influxdb/influxdb/issues/3968
```
Comment (corylanou) [inline cmd/influxd/run/server_test.go]: Is there anything unique about this test other than duplicate tags?  Also, the idiom I like to use is put a `skip` in my test struct, then the code isn't commented out, and if you do a verbose log, you can `t.Logf` out any skipped table tests.

Comment (dgnorton) [inline cmd/influxd/run/server_test.go]: It checks the results of a CQ with `group by *`.

**Comment by corylanou** [comment]
This might be beyond the scope of this PR, but how do we handle:

- A long running CQ or potentially a hung CQ?
- Can the same CQ run again if the previous one is still running?

Looks good. +1

File: cmd/influxd/run/server_test.go (hunk 5 above)
Comment (corylanou) [inline cmd/influxd/run/server_test.go]: Ok, that's what I figured, just wanted to check.

**Comment by dgnorton** [comment]
@corylanou 

* Long running - wait for it
* Hung - not sure.  Maybe ExecuteQuery times out?
* I don't think it can run again because the service waits for results.

I mocked enough of the interfaces in the CQ service tests that it should be relatively easy to add tests for those things but I don't think they're currently tested.

**Comment by shanquanqiang** [comment]
I want to add suffix for measurement names ,may by like that
INTO "1hPolicy".'$1_sss' FROM /.*/
is that supported？if it is not supported ,how to add suuffix for all measurement names?

**Pull request merged**